src/python/pants/engine/rules_test.py

```diff
@@ -692,16 +689,16 @@ async def d_from_a_and_suba(a: A, suba: SubA) -> D:  # type: ignore[return]
             self,
             dedent(
                 f"""\
-                Rules with errors: 3
+                Rules with errors: 2
 
                   {fmt_rule(a_from_c)}:
                     Was not reachable, either because no rules could produce the params or because it was shadowed by another @rule.
-                  {fmt_rule(b_from_d)}:
 
-                    No rule was available to compute D with parameter type SubA
                   {fmt_rule(d_from_a_and_suba, gets=[("A", "C")])}:
-
-                    No rule was available to compute A with parameter type SubA
+                    No rule was able to compute A.:
+                      @rule(C) -> A
+                pants.engine.rules_test:667:a_from_c
+                for SubA: Was unfulfillable.
```

Comment:
Regression in formatting. Ideally this would use the single line short representation used by `fmt_rule()`.

Reply:
Agreed: but this isn't a regression in this patch, as I suggested fixing it in #9740. Will fix it here though.

Reply:
Okay, sounds good. Thanks!

Reply:
Fixed in the most recent commit.